automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/terraredirect/RedirectSpec.scala

```diff
@@ -0,0 +1,86 @@
+package org.broadinstitute.dsde.firecloud.test.terraredirect
+
+import akka.http.scaladsl.model.Uri
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.firecloud.fixture.UserFixtures
+import org.broadinstitute.dsde.firecloud.page.methodrepo.MethodRepoPage
+import org.broadinstitute.dsde.workbench.config.UserPool
+import org.broadinstitute.dsde.workbench.fixture.{BillingFixtures, TestReporterFixture, WorkspaceFixtures}
+import org.broadinstitute.dsde.workbench.service.test.{CleanUp, WebBrowserSpec}
+import org.scalatest.{BeforeAndAfterAll, FreeSpec, Matchers}
+
+
+/**
+  * This spec asserts that a user who loads specific pages in the FireCloud UI is appropriately
+  * redirected to the Terra UI.
+  *
+  * As of this writing, redirects are disabled, because the redirects will break OTHER tests.
+  * To verify THIS test passes, you must do two things:
+  *   1) unignore any test you want to run, inside this file
+  *   2) set the UI's flag to enable redirects. You can do this by either or:
+  *     a) changing "terraRedirectsEnabled" to true in the UI instance's config.json
+  *     b) in the UI's config.cljs, change the terra-redirects-enabled function to return a hardcoded
+  *         value of true instead of returning (get @config "terraRedirectsEnabled" false)
+  */
+class RedirectSpec extends FreeSpec with BeforeAndAfterAll with Matchers with WebBrowserSpec
+  with UserFixtures with WorkspaceFixtures with BillingFixtures with CleanUp with LazyLogging with TestReporterFixture {
+
+  val user = UserPool.chooseAnyUser
+
+  // TODO: this test sends traffic from a FiaB or other test environment directly to the Terra dev server.
+  //  tests will fail if the dev server is unresponsive, and tests will cause additional load on the dev server.
+  //  is this acceptable?
+  private def isTerraUrl(url: String) = {
+    // should this use an environment-specific full hostname instead of contains()?
+    url.contains("terra.bio") ||
+    (url.contains("saturn") && url.contains("appspot.com"))
+  }
+
+  // negative test - ensure that Library/Methods Repo/whatever does NOT redirect
+  "A user who visits FireCloud should NOT be redirected to the same location in Terra for" - {
+
+    "methods repo" in {
+      implicit val authToken = user.makeAuthToken()
+      withWebDriver { implicit driver =>
+        withSignIn(user) { _ =>
```

Comment:
auto redirect could happen before action of opening methodRepoPage after signed in. there is a short delay after sign in.

Reply:
true ... but then `MethodRepoPage().open` would fail, and the test would fail, which is the desired result

Reply:
sorry for I'm asking stupid questions. I'm not clear on what should happens when:
1. user log in FC
2. on workspace list page, user doesn't immediately click on any tab.
3. scroll page up/down. Does auto-redirect happen?
4. user in Terra, user wants to go Method repo page. how does user get back to Method page?

Reply:
@aweng98 see my other comment for explanation of requirements. Certainly no redirect happens on *scroll*.

I don't understand your fourth question. If the user is in methods repo and clicks on a method, user will go to that method. Is that what you're asking?